eli5/formatters/image.py

```diff
@@ -0,0 +1,326 @@
+        Maximum opacity for each alpha value in the final alpha channel.
+    
+        See :func:`eli5.format_as_image` and :func:`cap_alpha` 
+        for more details on the ``alpha_limit`` parameter.
+
+    Returns
+    -------
+    None. *This function modifies ``image_array`` in-place.*
+    """
+    # get the alpha channel slice
+    if isinstance(starting_array, np.ndarray):
+        alpha = starting_array
+    else:
+        # take the alpha channel as is
+        alpha = image_array[:,:,3]
+    # set maximum alpha value
+    alpha = cap_alpha(alpha, alpha_limit)
+    # update alpha channel in the original image
+    image_array[:,:,3] = alpha
+    # TODO: optimisation?
+
+
+def cap_alpha(alpha_arr, alpha_limit):
+    # type: (np.ndarray, Union[None, float, int]) -> np.ndarray
+    """
+    Limit the alpha values in ``alpha_arr``
+    by setting the maximum alpha value to ``alpha_limit``.
+
+    Parameters
+    ----------
+    alpha_arr: numpy.ndarray
+        A rank 2 alpha channel numpy array, normalized to [0, 1] float values.
+    
+    alpha_limit : int or float, optional
+        A real between 0 and 1, representing the maximum alpha value.
+
+        If omitted, no capping is done, i.e. `alpha_limit = 1`.
+
+    Returns
+    -------
+    new_alpha : numpy.ndarray
+        Array with alpha values capped.
+    
+
+    :raises ValueError: if ``alpha_limit`` is outside the [0, 1] interval.
+    :raises TypeError: if ``alpha_limit`` is not float, int, or None.
+    """
+    if alpha_limit is None:
+        return alpha_arr
+    elif isinstance(alpha_limit, (float, int)):
+        if 0 <= alpha_limit <= 1:
+            new_alpha = np.minimum(alpha_arr, alpha_limit)
+            return new_alpha
+        else:
+            raise ValueError('alpha_limit must be' 
+                             'between 0 and 1 inclusive, got: %f' % alpha_limit)
+    else:
+        raise TypeError('alpha_limit must be int or float,' 
+                        'got: {}'.format(alpha_limit))
+
+
+def resize_over(heatmap, image, interpolation):
+    # type: (Image, Image, Union[None, int]) -> Image
+    """ 
+    Resize the ``heatmap`` image to fit over the original ``image``,
+    using the specified ``interpolation`` method.
+    
+    Parameters
+    ----------
+    heatmap : PIL.Image.Image
+        Heatmap that is to be resized.
+
+    image : PIL.Image.Image
+        The image whose dimensions will be resized to.
+
+    interpolation : int or None
+        Interpolation to use when resizing.
+
+        See :func:`eli5.format_as_image` for more details on the `interpolation` parameter.
+
+    Returns
+    -------
+    resized_image : PIL.Image.Image
+        A resized PIL image.
+    """
+    # PIL seems to have a much nicer API for resizing than scipy (scipy.ndimage)
+    # Also, scipy seems to have some interpolation problems: 
+    # https://github.com/scipy/scipy/issues/8210
+    spatial_dimensions = (image.height, image.width)
```

Comment:
I think PIL uses (width, height) order

Reply:
You are right.

Worth mentioning that, unlike Pillow, Keras uses (height, width) order.

Reply:
Thanks for the fix! Yeah most DL frameworks use this order, because of how images are stored in arrays (with height dimension first), I think this confusing.